.circleci/config.yml

```diff
@@ -59,71 +71,161 @@ commands:
         type: string
     steps:
       - checkout
+      - restore_cache:
+          key: -<< pipeline.parameters.cache_key_version >>-win-sys-{{ .Branch }}-<< parameters.py_version >>
       - run:
-          name: Preparing system
+          name: "Preparing environment - system"
           command: |
             choco install -y --no-progress miniconda3
             C:\tools\miniconda3\Scripts\conda.exe init powershell
             choco install -y --no-progress openssl javaruntime
       - run:
-          name: Preparing conda environment
+          name: "Preparing environment - Hydra"
           command: |
             conda create -n hydra python=<< parameters.py_version >> pywin32 -qy
             conda activate hydra
-            pip install nox dataclasses
+            pip install nox@git+git://github.com/theacodes/nox.git@8dcefbf8f59293bc1925dd13376dd6d3aa66e53a
+            pip install dataclasses
+      - save_cache:
+          key: -<< pipeline.parameters.cache_key_version >>-win-sys-{{ .Branch }}-<< parameters.py_version >>
+          paths:
+            - C:\tools\miniconda3
+
+
+jobs:
+  test_macos:
+    parameters:
+      py_version:
+        type: string
+    macos:
+      xcode: "10.0.0"
+    steps:
+      - macos:
+          py_version: << parameters.py_version >>
       - run:
-          name: Testing Hydra
-          no_output_timeout: 20m
+          name: "Testing Hydra"
+          no_output_timeout: 10m
           command: |
-            $env:NOX_PYTHON_VERSIONS=<< parameters.py_version >>
-            $env:PYTHONIOENCODING="utf_8"
+            export NOX_PYTHON_VERSIONS=<< parameters.py_version >>
             conda activate hydra
-            nox
-            exit $LASTEXITCODE
+            pip install nox@git+git://github.com/theacodes/nox.git@8dcefbf8f59293bc1925dd13376dd6d3aa66e53a
+            pip install dataclasses
+            nox -s lint test_core test_jupyter_notebooks -ts
 
 
-executors:
-  macos:
-    macos:
-      xcode: "10.0.0"
-  linux:
+  test_linux:
     parameters:
       py_version:
         type: string
     docker:
       - image: circleci/python:<< parameters.py_version >>
-  win: win/default
+    steps:
+      - linux:
+          py_version: << parameters.py_version >>
+      - run:
+          name: "Testing Hydra"
+          command: |
+            export NOX_PYTHON_VERSIONS=<< parameters.py_version >>
+            pip install nox@git+git://github.com/theacodes/nox.git@8dcefbf8f59293bc1925dd13376dd6d3aa66e53a
+            pip install dataclasses
+            nox -s lint test_core test_jupyter_notebooks -ts
 
 
-jobs:
-  test_macos:
+  test_win:
+    parameters:
+      py_version:
+        type: string
+    executor: win/default
+    steps:
+      - win:
+          py_version: << parameters.py_version >>
+      - run:
+          name: Testing Hydra
+          no_output_timeout: 20m
```

Comment:
inconsistent timeout. on purpose?

Reply:
Yea, this was the original value in the config file, I just copy pasted them :)

Reply:
but you did change timeouts in other locations in the circleci file.

Reply:
Could you point me to that?

I added extra jobs for test plugins, but in general I just put 10m for macos and 20m for windows to be consistent with the previous version.

Reply:
The no-output-timeouts of the previous version are irrelevant. you improved the speed a lot.
Make it 10m everywhere.